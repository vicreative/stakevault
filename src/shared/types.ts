import { NativeStackScreenProps } from '@react-navigation/native-stack';

export type RootStackParamList = {
  Welcome: undefined;
  Landing: undefined;
};

export type WelcomeScreenProps = NativeStackScreenProps<RootStackParamList, 'Welcome'>;
