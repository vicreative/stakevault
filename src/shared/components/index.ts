export { default as Button } from './Button/Button';
export { default as Logo } from './Logo/Logo';
