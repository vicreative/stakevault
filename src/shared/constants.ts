export const PHONE_REGEX = /^([0]{1}|\+?[234]{3})([7-9]{1})([0|1]{1})([\d]{1})([\d]{7})$/g;
