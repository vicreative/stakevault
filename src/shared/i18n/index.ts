const en = {
  welcome: {
    name: 'Stakevault',
    skip: 'Skip',
    login: 'Log in',
    signup: 'Sign up',
    welcomeText: 'Get a reward any time you make a winning bet!',
  },
  register: {
    title: 'Sign up',
    subText: 'Start winning bonuses now with StakeVault',
    signup: 'Create Account',
    loginLink: 'Already have an account? login',
  },
};

export { en };
