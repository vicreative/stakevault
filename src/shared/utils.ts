/**
 * Checks if a string is numeric in nature.
 * @param {String} str
 * @returns {Boolean}
 */
export function isNumeric(str: string): Boolean {
  return (
    !isNaN(str as any) && // use type coercion to parse the _entirety_ of the string (`parseFloat` alone does not do this)...
    !isNaN(parseFloat(str as any))
  ); // ...and ensure strings of whitespace fail
}
