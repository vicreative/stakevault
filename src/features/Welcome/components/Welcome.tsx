import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { Headline as Title } from 'react-native-paper';
import { Button } from 'shared/components';
import { en } from 'shared/i18n';
import theme from 'shared/config/theme/default';
import WelcomeIllustration from './WelcomeIllustration';
import { WelcomeScreenProps } from 'shared/types';
import { Header } from 'shared/components';

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.accent,
    width: '100%',
    height: '100%',
    paddingLeft: 16,
    paddingRight: 16,
  },
  illustrationContainer: {
    alignItems: 'center',
    height: '64%',
    paddingTop: 80,
    justifyContent: 'center',
  },
  illustrationText: {
    color: theme.colors.primary,
    fontSize: 24,
    textAlign: 'center',
    fontFamily: 'Montserrat_700Bold',
    paddingTop: 44,
    fontWeight: '600',
  },
  buttonContainer: {
    height: '16%',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  buttonContent: {
    width: '48%',
  },
});

function Welcome({ navigation }: WelcomeScreenProps) {
  return (
    <View style={styles.container}>
      <Header onSkip={() => navigation.navigate('Landing')} />
      <View style={styles.illustrationContainer}>
        <WelcomeIllustration />
        <Title style={styles.illustrationText}>{en.welcome.welcomeText}</Title>
      </View>
      <View style={styles.buttonContainer}>
        <View style={styles.buttonContent}>
          <Button mode="contained" onPress={() => {}} accessibilityLabel={en.welcome.login}>
            {en.welcome.login}
          </Button>
        </View>
        <View style={styles.buttonContent}>
          <Button
            mode="outlined"
            onPress={() => navigation.navigate('Register')}
            accessibilityLabel={en.welcome.signup}
          >
            {en.welcome.signup}
          </Button>
        </View>
      </View>
    </View>
  );
}

export default Welcome;
