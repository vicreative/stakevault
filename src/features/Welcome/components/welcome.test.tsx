import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import { en } from 'shared/i18n';
import Welcome from './Welcome';

afterEach(cleanup);

describe('Welcome', () => {
  it('should return a greeting', () => {
    const { welcomeText } = en.welcome;

    const { toJSON, getByText } = render(<Welcome />);

    const foundWelcomeTextElement = getByText(welcomeText);

    expect(foundWelcomeTextElement.props.children).toEqual(welcomeText);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should not find text that does not exist in the DOM', () => {
    const notFoundText = 'This text does not exist';

    const { queryByText } = render(<Welcome />);

    const notFoundTextElement = queryByText(notFoundText);

    expect(notFoundTextElement).toBeNull();
  });

  it('should find the skip button via text', () => {
    const text = en.welcome.skip;

    const { getByText } = render(<Welcome />);

    const foundButton = getByText(text);

    expect(foundButton.props.children).toEqual(text);
  });

  it('should find the login button via accessibilityLabel', () => {
    const accessibilityLabel = en.welcome.login;

    const { getByA11yLabel } = render(<Welcome />);

    const foundButton = getByA11yLabel(accessibilityLabel);

    expect(foundButton).toBeTruthy();
  });

  it('should find the sign up button via accessibilityLabel', () => {
    const accessibilityLabel = en.welcome.signup;

    const { getByA11yLabel } = render(<Welcome />);

    const foundButton = getByA11yLabel(accessibilityLabel);

    expect(foundButton).toBeTruthy();
  });
});
