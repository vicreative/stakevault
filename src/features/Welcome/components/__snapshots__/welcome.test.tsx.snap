// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should return a greeting 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1D3557",
      "height": "100%",
      "paddingBottom": 20,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1D3557",
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#A8DADC",
            "fontSize": 24,
            "opacity": 0,
          }
        }
      >
        Stakevault
      </Text>
      <RNSVGSvgView
        bbHeight={48}
        bbWidth={48}
        fill="none"
        focusable={false}
        height={48}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 48,
              "width": 48,
            },
          ]
        }
        width={48}
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx={24}
            cy={24}
            fill={4289256156}
            propList={
              Array [
                "fill",
              ]
            }
            r={24}
          />
          <RNSVGPath
            d="M11.356 8.43h28.098v2.81H12.878c-1.522 0-1.552 1.022-1.522 2.809v5.62c.077 1.494 0 2.926 1.522 2.926h22.01c4.566 0 4.566.703 4.566 4.566v7.727c0 2.927 0 4.683-4.566 4.683H8.546v-2.81h25.171c2.81 0 2.81-1.288 2.81-2.81v-5.853c.186-1.653 0-2.693-2.81-2.693H11.356c-2.81 0-2.777-1.763-2.81-5.737v-5.62c.057-3.124 0-5.619 2.81-5.619z"
            fill={4280104279}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M16.273 19.727h15.22v8.429h-15.22z"
            fill={4289256156}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGCircle
            cx={23.883}
            cy={24}
            fill={4280104279}
            propList={
              Array [
                "fill",
              ]
            }
            r={7.024}
          />
          <RNSVGEllipse
            cx={23.883}
            cy={23.883}
            fill={4289256156}
            propList={
              Array [
                "fill",
              ]
            }
            rx={4.098}
            ry={4.156}
          />
          <RNSVGPath
            d="M25.405 16.273h-2.81v-2.458c0-2.576 2.81-2.81 2.81 0v2.458zM25.405 31.71h-2.81v2.46c0 2.575 2.81 2.81 2.81 0v-2.46zM30.15 19.477l-1.987-1.987L29.9 15.75c1.822-1.82 3.974 0 1.987 1.987l-1.738 1.739zM30.15 28.754l-1.987 1.987L29.9 32.48c1.822 1.82 3.974 0 1.987-1.987l-1.738-1.739zM19.373 17.495l-1.987 1.986-1.739-1.738c-1.82-1.821 0-3.974 1.987-1.987l1.739 1.739zM19.373 30.746l-1.987-1.987-1.739 1.738c-1.82 1.822 0 3.974 1.987 1.987l1.739-1.738z"
            fill={4280104279}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#A8DADC",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "elevation": 0,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityLabel="Skip"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row-reverse",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": -4,
                  "marginRight": 12,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontSize": 16,
                    },
                  ]
                }
              >
                â–¡
              </Text>
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Skip
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "74%",
        "justifyContent": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 80,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={249}
      bbWidth={210}
      fill="none"
      focusable={false}
      height={249}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 249,
            "width": 210,
          },
        ]
      }
      width={210}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGGroup
          clipPath="prefix__clip0"
        >
          <RNSVGPath
            d="M33.114 35.782C12.532 48.696-2.046 71.896.234 93.763c2.33 21.817 21.524 42.347 28.117 68.224 6.593 25.878.645 57.189 11.656 73.117 10.957 15.982 38.992 16.673 61.836 9.647 22.91-6.975 40.663-21.667 60.4-34.827 19.736-13.112 41.419-24.737 46.518-41.509 5.099-16.721-6.398-38.601-18.896-54.912-12.448-16.376-25.984-27.16-39.968-42.248-13.983-15.087-28.465-34.432-48.996-42.79-20.579-8.363-47.211-5.643-67.787 7.317z"
            fill={4294048494}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M89.833 179.771l4.35-77.351M89.833 179.771l52.491-69.421M89.833 179.77l55.79-121.187"
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeMiterlimit",
              ]
            }
            stroke={4294890373}
            strokeLinecap={1}
            strokeMiterlimit={10}
            strokeWidth={5}
          />
          <RNSVGPath
            d="M179.803 46.731c5.72-15.685.423-32.006-11.83-36.452-12.253-4.447-26.823 4.664-32.543 20.35-5.719 15.685-.423 32.005 11.83 36.452 12.253 4.447 26.823-4.664 32.543-20.35z"
            fill={4293278022}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M121.183 80.715c1.164-16.65-8.437-30.882-21.444-31.787-13.007-.905-24.495 11.859-25.659 28.51-1.164 16.65 8.436 30.882 21.444 31.787 13.007.905 24.495-11.859 25.659-28.51z"
            fill={4292050009}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M183.101 94.137c8.119-16.643 3.289-35.675-10.787-42.508-14.076-6.834-32.068 1.118-40.187 17.76-8.118 16.643-3.289 35.675 10.787 42.509 14.076 6.834 32.068-1.118 40.187-17.761zM137.149 32.322c2.401-15.744-6.247-30.114-19.316-32.097-13.069-1.984-25.61 9.171-28.01 24.914-2.401 15.744 6.247 30.114 19.316 32.097 13.069 1.984 25.61-9.171 28.01-24.914z"
            fill={4289256156}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M144.448 73.83c4.494-14.777-.773-29.453-11.764-32.78-10.992-3.328-23.546 5.953-28.04 20.73-4.495 14.776.772 29.453 11.764 32.78 10.991 3.327 23.545-5.954 28.04-20.73z"
            fill={4282743709}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M95.478 108.321l2.7 6.325a1.15 1.15 0 01-1.2 1.592l-6.836-.841a1.155 1.155 0 01-.922-.692 1.145 1.145 0 01.142-1.142l4.149-5.485a1.152 1.152 0 011.967.243z"
            fill={4292050009}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M89.833 179.771L79.935 68.757M89.833 179.77l13.198-43.986 13.197-43.987"
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeMiterlimit",
              ]
            }
            stroke={4294890373}
            strokeLinecap={1}
            strokeMiterlimit={10}
            strokeWidth={5}
          />
          <RNSVGPath
            d="M117.206 91.878l1.572 6.694a1.142 1.142 0 01-.336 1.099 1.153 1.153 0 01-1.122.26l-6.599-1.996a1.151 1.151 0 01-.768-1.434c.059-.193.168-.368.315-.505l5.028-4.695a1.15 1.15 0 011.91.577z"
            fill={4294890373}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M144.096 109.186l-.48 6.852a1.152 1.152 0 01-1.799.874l-5.699-3.843a1.142 1.142 0 01-.506-1.034 1.15 1.15 0 01.647-.952l6.187-3.011a1.162 1.162 0 011.146.083 1.155 1.155 0 01.504 1.031z"
            fill={4285847006}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M80.228 72.094c15.604-1.253 27.048-17.224 25.56-35.674C104.3 17.97 90.444 4.03 74.84 5.282 59.236 6.535 47.793 22.506 49.28 40.956c1.489 18.45 15.344 32.39 30.948 31.138z"
            fill={4282743709}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M80.763 69.484l3.9 5.655a1.149 1.149 0 01-.297 1.594 1.156 1.156 0 01-.559.202l-6.853.565a1.156 1.156 0 01-1.24-1.054 1.15 1.15 0 01.106-.586l2.954-6.206a1.15 1.15 0 011.533-.554c.183.086.34.218.456.384z"
            fill={4294804329}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M62.16 234.888l65.684 13.023 11.693-58.698-65.684-13.022-11.693 58.697z"
            fill={4292775528}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M76.504 196.542l65.682 13.023-2.637-20.354-65.685-13.022 2.64 20.353zM24.56 227.428l37.606 7.456 11.693-58.698-37.606-7.456-11.693 58.698z"
            fill={4292050009}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M63.81 194.023l-37.535-7.442 10.057-17.837 37.532 7.445-10.054 17.834z"
            fill={4292775528}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M87.38 239.888l15.245 3.022 3.45-17.316-15.245-3.022-3.45 17.316z"
            fill={4292050009}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M70.918 212.172l22.958 4.552 2.518-12.641-22.958-4.552-2.519 12.641z"
            fill={4294890373}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M86.976 213.435l5.19 1.029 1.829-9.184-5.19-1.029-1.83 9.184zM74.379 203.261l12.968 2.571.348-1.749-12.967-2.571-.349 1.749zM73.78 206.254l12.967 2.571.349-1.749-12.968-2.571-.349 1.749z"
            fill={4294804329}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
        <RNSVGDefs>
          <RNSVGClipPath
            name="prefix__clip0"
          >
            <RNSVGPath
              d="M0 0h210v249H0z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGClipPath>
        </RNSVGDefs>
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 24,
                "letterSpacing": 0,
                "lineHeight": 32,
                "marginVertical": 2,
              },
              Object {
                "color": "#A8DADC",
                "fontFamily": "Montserrat_700Bold",
                "fontSize": 24,
                "fontWeight": "600",
                "paddingTop": 44,
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
    >
      Get a reward any time you make a winning bet!
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": "12%",
        "justifyContent": "space-between",
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      style={
        Object {
          "width": "48%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6200ee",
            "borderColor": "#A8DADC",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "elevation": 2,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityLabel="Log in"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Log in
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "48%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#A8DADC",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "elevation": 0,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityLabel="Sign up"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Sign up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
